### PR TITLE
[#3683] Use the new github review messages beside our review markers

### PR DESCRIPTION
**Pull request by adiroiban**
Scope
---------

This update our github hook server to be aware of the new github messages dedicated to the review process

Changes
-----------

How to test
-------------

reviewers: @brunogola @alibotean 

You can now use the general github review messages, without a in-text keyword to approve or request changes for a PR.

The review is still requested using the normal comment

Use this PR to check that all is ok

### CODE REVIEW

**Comment by adiroiban** [comment]
needs-review

**Comment by adiroiban** [comment]
needs-review

**Comment by brunogola** [comment]
it seems like it is not working properly, or am I using it wrong?

needs-changes

**Comment by adiroiban** [comment]
changes-approved

**Comment by brunogola** [comment]
still not working? 
needs-changes

**Comment by adiroiban** [comment]
needs-review

**Comment by adiroiban** [comment]
needs-review

File: chevah/github_hooks_server/tests/test_handler.py [hunk 1]
```diff
@@ -99,10 +122,13 @@ def test_issue_comment_no_pull(self):
...
 
         self.assertFalse(self.handler.trac.getTicket.called)
+        self.assertLog(
+            '[some] Not a comment on a pull request'
+            )
 
     def test_issue_comment_no_ticket(self):
         """
-        Noting happend is pull request title does not contain a ticket.
+        Noting happens is pull request title does not contain a ticket.
```
Comment (brunogola) [inline chevah/github_hooks_server/tests/test_handler.py]: this docstring is weird.. shouldnt it be **Nothing happens if** ...?

also, the test shows that something happens... there is a log message being triggered, but i'm not sure if it should be part of the specification / docstring

**Comment by adiroiban** [comment]
lets reset the workflow

needs-review

File: chevah/github_hooks_server/tests/test_handler.py [hunk 2]
```diff
@@ -272,20 +319,140 @@ def test_issue_comment_approved_still_reviewers(self):
...
+
+        self.handler.dispatch(event)
+
+        self.assertFalse(self.handler.trac.getTicket.called)
+        self.assertLog(
+            'Pull request has no ticket id in title: Some message.')
+
+    def test_pull_request_review_comment(self):
+        """
+        Nothing happens when the we got a simple review comment.
```
Comment (brunogola) [inline chevah/github_hooks_server/tests/test_handler.py]: there is an extra **the** there.

**Comment by adiroiban** [comment]
so... lets try again

needs-review

**Comment by adiroiban** [comment]
so... ticket is now in needs_changes...


let's request a review

needs-review

File: chevah/github_hooks_server/handler.py [hunk 3]
```diff
@@ -79,6 +79,63 @@ def closeTicket(self, commit):
...
+            # An needs changes review comment.
+            self._setNeedsChanges(ticket_id, user, body)
+        else:
+            # Just a simple comment.
+            # Do nothing
+            return
+
+    def _setNeedsChanges(self, ticket_id, user, body):
+        """
+        Set the ticket with `ticket_id` in needs changes state.
```
Comment (brunogola) [inline chevah/github_hooks_server/handler.py]: just adding a random internal comment

File: chevah/github_hooks_server/handler.py [hunk 4]
```diff
@@ -113,41 +167,22 @@ def issue_comment(self, event):
...
-                cc = ', '.join(reviewers)
-                ticket.requestMerge(comment=comment, cc=cc)
-            else:
-                comment = u'%s approved changes.\n\n%s' % (user, body)
-                ticket.update(
-                    attributes={'cc': ', '.join(remaining_reviewers)},
-                    comment=comment)
-            self._current_ticket = ticket
+        elif self._changesApproved(body):
+            self._setApproveChanges(ticket_id, user, body, reviewers)
```
Comment (brunogola) [inline chevah/github_hooks_server/handler.py]: another random internal comment

**Comment by adiroiban** [comment]
Thanks for the review.

Please check latest changes.

needs-review

File: chevah/github_hooks_server/tests/test_handler.py [hunk 5]
```diff
@@ -272,20 +319,140 @@ def test_issue_comment_approved_still_reviewers(self):
...
+        self.handler.dispatch(event)
+
+        self.handler.trac.getTicket.assert_called_once_with(42)
+        ticket.requestChanges.assert_called_once_with(
+            comment=u'tu requested changes to this ticket.\n\nanything here.')
+
+    def test_pull_request_review_approved_last(self):
+        """
+        When the review was approved and no other reviewers are expected,
+        the ticket is set into the needs-merge state.
```
Comment (brunogola) [inline chevah/github_hooks_server/tests/test_handler.py]: two more internal messages

File: chevah/github_hooks_server/tests/test_handler.py [hunk 6]
```diff
@@ -421,6 +588,9 @@ def test_getRemainingReviewers_not_found(self):
 
         self.assertEqual(['ala', 'bala'], result)
 
+        self.assertLog(
+            'Current user "popa" not in the list of reviewers ala, bala')
+
```
Comment (brunogola) [inline chevah/github_hooks_server/tests/test_handler.py]: last one

File: chevah/github_hooks_server/handler.py [hunk 7]
```diff
@@ -79,6 +79,63 @@ def closeTicket(self, commit):
             )
         self._current_ticket = ticket
 
+    def pull_request_review(self, event):
```
Comment (alibotean) [inline chevah/github_hooks_server/handler.py]: String test

File: chevah/github_hooks_server/handler.py [hunk 8]
```diff
@@ -79,6 +79,63 @@ def closeTicket(self, commit):
...
+        comment = u'%s requested changes to this ticket.\n\n%s' % (
+            user, body)
+        ticket.requestChanges(comment=comment)
+        self._current_ticket = ticket
+
+    def _setApproveChanges(self, ticket_id, user, body, reviewers):
+        """
+        Update the ticket with `ticket_id` as approved.
+        """
+        ticket = self.trac.getTicket(ticket_id)
```
Comment (alibotean) [inline chevah/github_hooks_server/handler.py]: This is no good :smile:

File: chevah/github_hooks_server/tests/test_handler.py [hunk 9]
```diff
@@ -202,6 +240,10 @@ def test_issue_comment_request_changes(self):
             content['comment']['body']
             )
         rc.assert_called_once_with(comment=comment)
+        self.assertLog(
+            '[some][12] New comment from somebody with reviewers []\n'
```
Comment (alibotean) [inline chevah/github_hooks_server/tests/test_handler.py]: Hmmm

**Pull request merged**